### ISSUE TITLE
Double construction of `frameBuffer`

### ISSUE THREAD

**Issue by EWNB** [Harbys/pico-ssd1306#12]
I think the line `this->frameBuffer = FrameBuffer();` ([ssd1306.cpp:19](https://github.com/Harbys/pico-ssd1306/blob/e6438aa1bb25b414558b09a718a09dcb415cb9cd/ssd1306.cpp#L19)) in the `SSD1306` constructor is both redundant and can cause problems.

What I think is happening is that `frameBuffer` is already initialized by the time the body of the `SSD1306` constructor is entered. The line `this->frameBuffer = FrameBuffer();` replaces the address to first `FrameBuffer` with the address of a second `FrameBuffer` that is constructed in the scope of the `SSD1306` constructor. When the `SSD1306` constructor is exited, the second `FrameBuffer` is destructed and it's buffer is freed. This leaves the just-constructed `SSD1306` object with a `frameBuffer` member pointing to a `FrameBuffer` object with a buffer that has been freed, causing it to corrupt other data when that memory is allocated to something else.

On member initialization, from <https://en.cppreference.com/w/cpp/language/constructor>:
> Before the compound statement that forms the function body of the constructor begins executing, initialization of all direct bases, virtual bases, and non-static data members is finished.

As a test, I added print statements to the constructor and destructor in `FrameBuffer.cpp`:
```cpp
#include <stdio.h>

FrameBuffer::FrameBuffer() {
    this->buffer = new unsigned char[FRAMEBUFFER_SIZE];
    printf("FrameBuffer() (%p) new buffer: %p\n", this, this->buffer);
}

FrameBuffer::~FrameBuffer() {
    printf("~FrameBuffer() (%p) delete buffer: %p\n", this, this->buffer);
    delete[] this->buffer;
}
```

I then modified the `basic_text` example to add in some more print statements and to match the hardware I have:
```cpp
#include <cstdio>
#include "pico/stdlib.h"
#include "pico-ssd1306/ssd1306.h"
#include "pico-ssd1306/textRenderer/TextRenderer.h"
#include "hardware/i2c.h"

// Use the namespace for convenience
using namespace pico_ssd1306;

int main(){
    // Initialise usb serial for debugging and give our terminal monitor
    // a bit of time to connect.
    stdio_init_all();
    sleep_ms(2000);
    printf("comms test\n");

    // Init i2c0 controller
    i2c_init(i2c0, 1000000);
    // Set up pins 12 and 13
    gpio_set_function(PICO_DEFAULT_I2C_SDA_PIN, GPIO_FUNC_I2C);
    gpio_set_function(PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C);
    gpio_pull_up(PICO_DEFAULT_I2C_SDA_PIN);
    gpio_pull_up(PICO_DEFAULT_I2C_SCL_PIN);

    // If you don't do anything before initializing a display pi pico is too fast and starts sending
    // commands before the screen controller had time to set itself up, so we add an artificial delay for
    // ssd1306 to set itself up
    sleep_ms(250);

    // Create a new display object at address 0x3D and size of 128x64
    printf("before SSD1306()\n");
    SSD1306 display = SSD1306(i2c0, 0x3C, Size::W128xH32);
    printf("after SSD1306()\n");
    printf("display.frameBuffer.get(): %p\n", display.frameBuffer.get());

    // Here we rotate the display by 180 degrees, so that it's not upside down from my perspective
    // If your screen is upside down try setting it to 1 or 0
    display.setOrientation(0);

    // Draw text on display
    // After passing a pointer to display, we need to tell the function what font and text to use
    // Available fonts are listed in textRenderer's readme
    // Last we tell this function where to anchor the text
    // Anchor means top left of what we draw
    drawText(&display, font_12x16, "TEST text", 0 ,0);

    // Send buffer to the display
    display.sendBuffer();
    printf("end\n");
}
```
Note that I also made `frameBuffer` `public` so I could call `display.frameBuffer.get()` in `main()`.

The resulting output was:
```log
19:24:20:727 -> comms test
19:24:20:977 -> before SSD1306()
19:24:20:980 -> FrameBuffer() (20041FE4) new buffer: 200035E0
19:24:20:984 -> FrameBuffer() (20041F9C) new buffer: 200039E8
19:24:20:988 -> ~FrameBuffer() (20041F9C) delete buffer: 200039E8
19:24:21:003 -> after SSD1306()
19:24:21:004 -> display.frameBuffer.get(): 200039E8
19:24:21:018 -> end
19:24:21:020 -> ~FrameBuffer() (20041FE4) delete buffer: 200039E8
```

This output seems to show two `FrameBuffer` objects being created, and the second freeing its buffer before we get back to `main()`. Further, the address of that freed buffer is what `display.frameBuffer.get()` returns so the `SSD1306` will use that freed buffer, clobbering anything written there after it is reallocated.

The fix seems simple. Delete `this->frameBuffer = FrameBuffer();` from `ssd1306.cpp`.

**Comment by EWNB** [comment]
This is my first attempt at reporting an issue and making a pull request on GitHub, so please let me know if there's anything I've missed or done incorrectly.

**Comment by Harbys** [comment]
Yea, you're right. It must have been an leftover from early development when frameBuffer was supposed to be a null pointer and initialized along with the display in constructor. Thanks for catching this.

fixed in #13

**Comment by Harbys** [comment]
> This is my first attempt at reporting an issue and making a pull request on GitHub, so please let me know if there's anything I've missed or done incorrectly.

All good with an exceptional amount of details. 👍
